src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/AutomaticEndAppealForNonPaymentEaHuTrigger.java

```diff
@@ -52,7 +52,44 @@ public boolean canHandle(
 
         return callback.getEvent() == Event.SUBMIT_APPEAL
                && (remissionType.isPresent() && remissionType.get() == RemissionType.NO_REMISSION)
-               && (appealType.isPresent() && (appealType.get() == AppealType.EA || appealType.get() == AppealType.HU));
+               && (appealType.isPresent() && (appealType.get() == AppealType.EA || appealType.get() == AppealType.HU))
+                && checkScenario(asylumCase);
+    }
+
+    private boolean checkScenario(AsylumCase asylumCase) {
+        Optional<YesOrNo> isDetained = Optional.of(asylumCase.read(APPELLANT_IN_DETENTION, YesOrNo.class).orElse(YesOrNo.NO));
+        Optional<YesOrNo> isAcceleratedDetainedAppeal = Optional.of(asylumCase.read(IS_ACCELERATED_DETAINED_APPEAL, YesOrNo.class).orElse(YesOrNo.NO));
+        Optional<AppealType> appealType = asylumCase.read(APPEAL_TYPE, AppealType.class);
+        Optional<YesOrNo> isAgeAssessmentAppeal = Optional.of(asylumCase.read(AGE_ASSESSMENT, YesOrNo.class).orElse(YesOrNo.NO));
+
+        if (isDetained.equals(Optional.of(YesOrNo.YES))) {
+            if (isAgeAssessmentAppeal.equals(Optional.of(YesOrNo.YES))) {
+                return true;
+            } else if (isAcceleratedDetainedAppeal.equals(Optional.of(YesOrNo.NO))
+                    && checkAppealTypeIsHuEaEu(appealType)) {
+                return true;
+            } else {
+                return false;
+            }
+        } else if (isDetained.equals(Optional.of(YesOrNo.NO))) {
+            if (isAgeAssessmentAppeal.equals(Optional.of(YesOrNo.NO))) {
+                return true;
+            } else if (checkAppealTypeIsHuEaEu(appealType)) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+
+        return false;
+    }
+
+    private boolean checkAppealTypeIsHuEaEu(Optional<AppealType> appealType) {
+        String appealTypeName = appealType.get().name();
+
+        return appealTypeName.equals(AppealType.HU.name())
+                || appealTypeName.equals(AppealType.EA.name())
+                || appealTypeName.equals(AppealType.EU.name());
```

Comment:
Why EU?

Reply:
In fact, the check for HU and EA is redundant, as it is already checked previously a few lines above. So only the check for EU is needed, as per the A/C